### ISSUE TITLE
lint: newline in log

### ISSUE THREAD

**Issue by cristaloleg** [go-critic/go-critic#557]
Name: `logNewline`
Before:
```go
log.Infof("oh it's a %s\n", foo)
```
After:
```go
log.Infof("oh it's a %s", foo)
```

Looks quite opinionated 😉

**Comment by Quasilyte** [comment]
SGTM.

**Comment by snargleplax** [comment]
Should this warn on all newlines in logs, or only at the end of the string?

**Comment by cristaloleg** [comment]
@snargleplax firstly, thanks for the contributions, much appreciated.

Regarding your question: I think it's a good thing to make it configurable, see `unnamedResult` checker for a checker's config https://github.com/go-critic/go-critic/pull/446/files

**Comment by snargleplax** [comment]
Okay, thanks. Next question, is there a particular logging library interface that's meant to be targeted here? I'd have assumed the standard library log package, but it does not define `Infof`.

**Comment by cristaloleg** [comment]
@snargleplax  `Infof` was a probably silly example. We can start with functions like: Log, Print, Debug, Fatal, Trace, Error, Warn, Panic. With such params: `format string, args ...interface{}`). Should we catch only functions with `f` suffix - I dunno. @quasilyte

**Comment by quasilyte** [comment]
We can start from a limited implementation that only handles standard log package.

**Comment by cristaloleg** [comment]
I'm proposing to close it 👀
